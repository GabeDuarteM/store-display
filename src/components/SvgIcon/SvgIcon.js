import React from 'react'

import PropTypes from 'prop-types'
import injectSheet from 'react-jss'
import cn from 'classnames'

export const styles = {
  svgIcon: {
    height: ({ height }) => height,
    width: ({ width }) => width,
    fill: ({ color }) => color,
    display: 'block',
  },
}

// eslint-disable-next-line no-unused-vars
const SvgIcon = ({
  classes,
  src,
  height,
  width,
  color,
  className,
  ...rest
}) => (
  <span
    className={cn(classes.svgIcon, className)}
    // eslint-disable-next-line react/no-danger
    dangerouslySetInnerHTML={{ __html: src }}
    {...rest}
  />
)

export const propTypes = {
  /** @ignore */
  classes: PropTypes.object.isRequired,
  /** If passed, the className will be applied to the root of the component */
  className: PropTypes.string,
  /** A string containing the content of the svg */
  src: PropTypes.string.isRequired,
  /** The height of the svg */
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** The width of the svg */
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** The color to use as the fill of the svg */
  color: PropTypes.string,
}
export const defaultProps = {
  height: 'auto',
  width: 'auto',
  color: null,
  className: null,
}

SvgIcon.propTypes = propTypes
SvgIcon.defaultProps = defaultProps

export default injectSheet(styles)(SvgIcon)
