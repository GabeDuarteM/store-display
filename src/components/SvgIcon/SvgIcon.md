```js
const downArrow = require('../../assets/down-arrow.svg')
;<SvgIcon width={10} src={downArrow} />
```
