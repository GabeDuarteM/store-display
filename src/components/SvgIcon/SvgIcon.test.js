import React from 'react'

import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import downArrow from '../../assets/down-arrow.svg'
import SvgIcon, { styles, propTypes, defaultProps } from './SvgIcon'

const createWrapper = (func, optionalProps) =>
  func(<SvgIcon src={downArrow} {...optionalProps} />)

const createWrapperShallow = optionalProps =>
  createWrapper(shallow, optionalProps)

const createWrapperMount = optionalProps => createWrapper(mount, optionalProps)

describe('COMPONENT: <SvgIcon />', () => {
  it('should render without crashing with default props', () => {
    createWrapperMount()
  })

  it('should render without crashing with custom props', () => {
    createWrapperMount({ width: 10, height: 25, color: '#e7e7e7' })
  })

  it("should match component's snapshot", () => {
    const wrapper = createWrapperShallow()
    expect(toJson(wrapper.dive())).toMatchSnapshot()
  })

  it("should match component's style snapshot", () => {
    expect(styles).toMatchSnapshot()
  })

  it("should match component's propTypes snapshot", () => {
    expect(propTypes).toMatchSnapshot()
  })

  it("should match component's defaultProps snapshot", () => {
    expect(defaultProps).toMatchSnapshot()
  })

  it('should apply whatever className that is passed as prop', () => {
    const className = 'customClassname'
    const wrapper = createWrapperShallow({ className })
    const hasCustomClassnameClass = wrapper.hasClass(className)
    expect(hasCustomClassnameClass).toBeTruthy()
  })

  it('should set the dangerouslySetInnerHTML as whatever is passed as the src prop', () => {
    const src = '<div>test</div>'
    const wrapper = createWrapperShallow({ src })
    const hasCorrectSrc = wrapper.dive().prop('dangerouslySetInnerHTML')
    expect(hasCorrectSrc).toEqual({ __html: src })
  })
})
