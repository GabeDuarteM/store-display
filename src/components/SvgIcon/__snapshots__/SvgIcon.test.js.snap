// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT: <SvgIcon /> should match component's defaultProps snapshot 1`] = `
Object {
  "className": null,
  "color": null,
  "height": "auto",
  "width": "auto",
}
`;

exports[`COMPONENT: <SvgIcon /> should match component's propTypes snapshot 1`] = `
Object {
  "className": [Function],
  "classes": [Function],
  "color": [Function],
  "height": [Function],
  "src": [Function],
  "width": [Function],
}
`;

exports[`COMPONENT: <SvgIcon /> should match component's snapshot 1`] = `
<span
  className="SvgIcon-svgIcon SvgIcon-svgIcon"
  dangerouslySetInnerHTML={
    Object {
      "__html": "",
    }
  }
/>
`;

exports[`COMPONENT: <SvgIcon /> should match component's style snapshot 1`] = `
Object {
  "svgIcon": Object {
    "display": "block",
    "fill": [Function],
    "height": [Function],
    "width": [Function],
  },
}
`;
